api/src/documents.py

```diff
@@ -323,17 +324,38 @@ def latest_documents(scroll_id: Optional[str]=None, size: int = 10):
 @router.get(
     "/documents", response_model=DocumentSchema.DocumentListResponse
 )
-def list_documents(scroll_id: Optional[str]=None, size: int = 10):
+def list_documents(
+    scroll_id: Optional[str] = None,
+    size: int = 10,
+    sort_by: Optional[str] = Query(None, description="Field to sort by"),
+    order: Optional[str] = Query(None, description="Order to sort by")
+):
     """
     Retrieves all stored documents, regardless of files uploaded or processed.
+    Optional `sort_by` and `order` params return sorted results.
     """
 
+    if order not in ["asc", "desc"]:
+        raise HTTPException(status_code=400, detail="Invalid sort order. Must be either 'asc' or 'desc'")
+
+    valid_fields = ["title", "publisher", "description", "creation_date"]
```

Comment:
On the API side, we should probably allow sorting by any fields (even if the UI will only sort by these). Unless there are issues with arbitrary `sort_by`. If so, let's talk about those. I can help!